cpp/src/join/joining.cu

```diff
@@ -533,6 +534,70 @@ gdf_error join_call_compute_df(
     return GDF_SUCCESS;
   }
 
+
+  //if the inputs are nvcategory we need to make the dictionaries comparable
```

Comment:
I think this whole code-block should go inside an `if` check that checks for the existence of a `GDF_STRING_CATEGORY`, otherwise you're doing a bunch of stuff that's unnecessary without a `GDF_STRING_CATEGORY` column.

You can do this very cleanly using `std::any`:

```
bool const at_least_one_category_column{ std::any( left_join_cols, left_join_cols + num_join_cols, [](gdf_column * col){ return col->dtype == GDF_STRING_CATEGORY; } };